tools/otsdb-dump-restore/src/kilda/tsdb_dump_restore/dump.py

```diff
@@ -69,11 +77,19 @@ def main(opentsdb_endpoint, time_start, **options):
     http_session.mount('http://', HTTPAdapter(max_retries=retries))
 
     client = stats_client.OpenTSDBStatsClient(http_session, opentsdb_endpoint)
-
     all_metrics_iterator = stats_client.OpenTSDBMetricsList(http_session, opentsdb_endpoint, prefix=prefix)
-    for metric in all_metrics_iterator:
-        dump(statistics, client, dump_frame, dump_dir, metric, query_frame_size, need_remove_meta=need_remove_meta)
 
+    concurrent_dump(all_metrics_iterator, statistics, client, dump_frame, dump_dir, query_frame_size, need_remove_meta, concurrent_executors)
+
+def concurrent_dump(all_metrics_iterator, statistics, client, dump_frame, dump_dir, query_frame_size, need_remove_meta, concurrent_executors):
+    with concurrent.futures.ThreadPoolExecutor(max_workers=concurrent_executors) as executor:
+        futures = []
+        for metric in all_metrics_iterator:
+            futures.append(executor.submit(dump, statistics, client, dump_frame, dump_dir, metric, query_frame_size, need_remove_meta))
+        # concurrent.futures.wait(futures)
+        concurrent.futures.as_completed(futures)
+        # for _ in concurrent.futures.as_completed(futures):
+        #     print("OK")
```

Comment:
This is for debugging or why it is here?

Reply:
This is just here because I forgot to remove it 😬. I'll remove it